src/Engines/MeilisearchEngine.php

```diff
@@ -378,7 +373,26 @@ public function deleteIndex($name)
      */
     public function deleteAllIndexes()
     {
-        return $this->meilisearch->deleteAllIndexes();
+        $tasks = [];
+        $offset = 0;
```

Comment:
Final thought on this $offset,

This variable becomes useless, also, creating a new instance of $query isn't necessary anymore.

Here is a proposition of something possibly simpler

```php
public function deleteAllIndexes()
{
    $tasks = [];
    $batchSize = 20;

    $query = new IndexesQuery();
    $query->setLimit($batchSize);

    do {
        $indexes = $this->meilisearch->getIndexes($query);

        foreach ($indexes->getResults() as $index) {
            $tasks[] = $index->delete();
        }
    }
    while ($indexes->count() >= $batchSize);

    return $tasks;
 }
 ```

Reply:
I thought about this a bit more and came to the conclusion that either of our solutions don't work correctly.

Index deletion is an async task, and the method will not correctly work as meili's tasks queue may not have finished deleting the task when we hit the next iteration in the loop. So the code will try to delete the same index multiple times and spam the meili task queue.

I think it would be better to just set a very high batch size and just take what ever is returned on the first page? what do you think @brunoocasali?

ping @driesvints

Reply:
The same light comes to me this morning..

Are the order of index guaranteed ? either name, or creation date ?
If so, a way to address the async issue would be, to get the total number of index first (using `$indexes->getTotal()`). Then crawl page and delete per page but from the last page to the first.

By doing this, whenever the deletion is completed or not at next batch, you should not overlap anymore.

Maybe this is completely overkill, and just setting a very high batch size is the best way to go.

Reply:
Hi there, I kind of went with my first solution just fetching all indexes first and when we have all of them in memory I iterate over them.

A bigger batch size makes sense. I think the (deleteAllIndexes) feature as is, probably is completely sufficient.

Reply:
Thanks @brunoocasali for clearing this up. I went with high batch number that should be sufficient. Hopefully this is something that a dedicated endpoint on meilisearch will handle 👍